### PR TITLE
Begin unit tests

### PR DESCRIPTION
**Pull request by mc18g13**

### CODE REVIEW

File: test/database_for_testing.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+from boxwise_flask.app import db
+
+
+def with_test_db(dbs: tuple):
+    def decorator(func):
+        @wraps(func)
+        def test_db_closure(*args, **kwargs):
+            test_db = db.database
+            with test_db.bind_ctx(dbs):
```
Comment (pylipp) [inline test/database_for_testing.py:11]: TIL about `bind_ctx`, sweet! :+1:

File: test/test_app.py [hunk 2]
```diff
@@ -1,29 +1,66 @@
...
-    db.init_app(app)
+    """test client to setup app"""
+    app.config["DATABASE"] = "sqlite:///:memory:"
 
     with app.test_client() as client:
         yield client
 
-    os.close(db_fd)
-    os.remove(db_filepath)
+    db.init_app(app)
```
Comment (pylipp) [inline test/test_app.py:22]: Is it intended that this is called after the block before? I thought that the `yield` statement would block execution, meaning that `db.init_app()` is never called when the client fixture is used.

Comment (mc18g13) [reply test/test_app.py:22]: I was very confused about this as well. it seems to work either side of the block.

File: test/auth_for_testing.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        "username": auth0_username,
+        "password": auth0_password,
+    }
+
+    response = requests.post(url, json=parameters, headers=headers).text
+    responseDICT = json.loads(response)
+    return responseDICT["access_token"]
+
+
+@memoize  # memoize code from: https://stackoverflow.com/a/815160
```
[the remark this answers is not in the document]
Comment (mc18g13) [reply test/auth_for_testing.py:43]: Yeah that's suppose to be the idea. I've sort of put my trust in stack overflow here.

**Comment by HaGuesto** [comment]
@mc18g13 can you give a reasoning why you replaced the `FlaskDB` helper from `playhous.flask_utils` by your own database_wrapper? At the end most of the code is copy/pasted from the helper.

**Comment by mc18g13** [comment]
Creating tables requires a call to connect as well as when the end point is hit. The wrapper just allows connect not to throw exceptions when another call to connect is made. If you close a connection to an in memory database I think it deletes it. I'd like to get this working with temp files again and try and close it but couldn't get it going.

**Comment by mc18g13** [comment]
> @mc18g13 can you give a reasoning why you replaced the `FlaskDB` helper from `playhous.flask_utils` by your own database_wrapper? At the end most of the code is copy/pasted from the helper.

@HaGuesto 
Creating tables requires a call to connect as well as when the end point is hit. The wrapper just allows connect not to throw exceptions when another call to connect is made. If you close a connection to an in memory database I think it deletes it. I'd like to get this working with temp files again and try and close it but couldn't get it going.

**Comment by HaGuesto** [comment]
[Write basic back-end tests to ensure everything is running correctly #flask #python #medium](https://trello.com/c/zttDLyPa/566-write-basic-back-end-tests-to-ensure-everything-is-running-correctly-flask-python-medium)

**Pull request merged**